python/ray/util/spark/cluster_init.py

```diff
@@ -211,8 +222,41 @@ def shutdown(self, cancel_background_job=True):
                 _logger.warning(
                     "An Error occurred during shutdown of ray head node: " f"{repr(e)}"
                 )
+            finally:
+                _try_clean_temp_dir_at_exit(
+                    process=self.head_proc,
+                    collect_log_to_path=os.environ.get(
+                        RAY_ON_SPARK_COLLECT_LOG_TO_PATH
+                    ),
+                    temp_dir=self.temp_dir,
+                )
```

Comment:
* It can only clean clean temp dir data on spark driver node, for temp dir data generated by ray worker node, it cannot clean them.

* It causes race condition because the "head_proc" also installs exit hook for executing `_try_clean_temp_dir_at_exit`

Reply:
To address this, I recommend for global mode, we do temp cleanup at the beginning of `start_ray_node.py` process (clean the temp dir of last shutdown global ray cluster)

Reply:
Looks like this is not restricted to global mode?

Reply:
Wait I think inside start_ray_node.py, when it's terminated it calls `try_clean_temp_dir_at_exit` to clean up the tempdir

Reply:
> Wait I think inside start_ray_node.py, when it's terminated it calls try_clean_temp_dir_at_exit to clean up the tempdir

Yes we can use this approach